aibolit/patterns/if_return_if_detection/if_detection.py

```diff
@@ -24,9 +11,15 @@ class CountIfReturn:
     def __init__(self):
         pass
 
-    def value(self, filename: str):
-        ''''''
-        tree = newJavalangImproved(filename)
-        if_decls = tree.filter([javalang.tree.IfStatement])
-
-        return if_decls
+    def value(self, filename: str) -> List[int]:
+        detected_lines = []
+        p = JavaPackage(filename)
+        for class_name in p.java_classes:
+            tree = p.java_classes[class_name]
+            for index, each_object in enumerate(tree.nodes_by_type(ASTNodeType.IF_STATEMENT)):
+                all_childs = list([i for i in tree.tree.succ[each_object]])
```

Comment:
You don't need the outer `list`. list comprehension do this job.

Reply:
got it